File: test/react-native/features/fixtures/ios-module/Scenarios/UserNativeClientScenario.m
#import "UserNativeClientScenario.h"

@implementation UserNativeClientScenario

- (void)run: (RCTPromiseResolveBlock)resolve
     reject:(RCTPromiseRejectBlock)reject {
  [Bugsnag setUser:@"123" withEmail:@"bug@sn.ag" andName:@"Bug Snag"];
  @throw [[NSException alloc] initWithName:@"NSException" reason:@"UserNativeClientScenario" userInfo:nil];
}

@end

File: test/react-native/features/fixtures/reactnative/scenarios/UserNativeClientScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag

class UserNativeClientScenario(context: Context): Scenario(context) {

    override fun run() {
        super.run()
        Bugsnag.setUser("123", "bug@sn.ag", "Bug Snag")
        throw generateException()
    }
}
